### PR TITLE
feat: add custom comparator

### PR DESCRIPTION
**Pull request by lyl156**
Refactor func to sort key and add custom comparator. #373

### CODE REVIEW

File: db.go [hunk 1]
```diff
@@ -192,6 +192,67 @@ type (
 	BucketMetasIdx map[string]*BucketMeta
 )
 
+func (e Entries) Len() int { return len(e) }
```
Comment (elliotchenzichang) [inline db.go:195]: Would suggest this part of code move to another seperated file so that make db.go focus its own logic.

**Review by elliotchenzichang — COMMENTED** [review]
General LGTM, left a thin comment. Thanks!

File: db.go (hunk 1 above)
Comment (lyl156) [reply db.go]: I moved `Entries` to `entry.go`

**Review by elliotchenzichang — COMMENTED** [review]
LGTM! Thanks for this great contribution! 
BTW, this implementation is very elegant.

**Comment by elliotchenzichang** [comment]
@lyl156 could you help change the readme.md to add an example and explaination of this exciting feature? Thanks in advance.

**Comment by lyl156** [comment]
> @lyl156 could you help edit the readme.md to add an example and explaination of this exciting feature? Thanks in advance.

Please read #391.

**Pull request merged**